src/vs/workbench/api/common/extHostChatAgents2.ts

```diff
@@ -389,8 +391,13 @@ export class ExtHostChatAgents2 extends Disposable implements ExtHostChatAgentsS
 
 			stream = new ChatAgentResponseStream(agent.extension, request, this._proxy, this._commands.converter, sessionDisposables);
 
+			const extRequest = typeConvert.ChatAgentRequest.to(request, location);
+			if (request.userSelectedModelId) {
+				extRequest.userSelectedModel = await this._languageModels.getLanguageModelByIdentifier(agent.extension, request.userSelectedModelId);
```

Comment:
Maybe move the proposed API check down into this (instead of deleting it from the request)